File: 50 Coding Challenge Part I/TypeScript/Challenge18.ts
// Challenge 18: Print the first 100 prime numbers

// function prints th first nPrime numbers
function printPrime(nPrime: number): void {
    let n: number = 0;
    let i: number = 2;

    while (n < nPrime){
        if (isPrime(i)){
            console.log(`--> ${i}`)
            // n should only increment if i is a prime
            n++;
        }
        i++;
    }
}

// function determines if a number is prime
function isPrime(num: number): boolean {
    if (num < 2){
        return false;
    }

    if (num == 2){
        return true;
    }

    let maxDiv: number = Math.sqrt(num);

    for (let i = 2; i <= maxDiv; i++){
        if (num % i == 0){
            return false;
        }
    }

    return true;
}


// Driver Method
printPrime(100)

File: 50 Coding Challenge Part I/dart/lib/challenge18.dart
/*  
  Coding Challenge 18: Print the first 100 prime numbers

  @author: mofirojean
*/

// Functions prints the first nth prime numver
import 'package:dart/challenge16.dart';

void printPrimes(int nPrimes) {
  int n = 1;
  int i = 2;

  while (n < nPrimes) {
    if (isPrime(i)) {
      print('$n ---> $i');
      n++;
    }
    i++;
  }
}
